Sources/MapboxNavigation/SpriteRepository.swift

```diff
@@ -156,6 +140,27 @@ class SpriteRepository {
         return imageCache.image(forKey: "Legacy")
     }
     
+    func shieldColor(from shieldTextColor: String) -> UIColor {
+        switch shieldTextColor {
+        case "black":
+            return .black
+        case "blue":
+            return .blue
+        case "green":
+            return #colorLiteral(red: 0, green: 0.5, blue: 0, alpha: 1)
```

Comment:
All of these colors should be specified by the Style, not hard-coded here in the sprite repository. A different map style could have a completely different color scheme, which would lead to a spritesheet with a different color scheme too. We wouldn’t want to composite text onto the image that uses a mismatching color.

We can split out DayStyle/NightStyle modifications as tail work, since it might require investigating the specific colors used by the most popular map styles, but let’s make sure we have a plan for how to expose the color settings to Style and how to clear the cache when the UI style changes between day and night.

Reply:
Thanks for the suggestion, I'll add the enumeration of each road shield text color corresponding UIColor in the Style. So we could work on the different color style. Right now the `WayNameLabel` would do the work to match the shield text color of the  road shield UIColor.